src/main/groovy/com/ullink/AssemblyInfoVersionPatcher.groovy

```diff
@@ -61,10 +61,20 @@ class AssemblyInfoVersionPatcher extends ConventionTask {
     void run() {
         getPatchedFiles().each {
             logger.info("Replacing version attributes in $it")
-            replace(it, 'AssemblyVersion', getVersion())
-            replace(it, 'AssemblyFileVersion', getFileVersion())
-            replace(it, 'AssemblyInformationalVersion', getInformationalVersion())
-            replace(it, 'AssemblyDescription', getAssemblyDescription())
+            //only change the values if they specified in this build script.
+            //if the parameters are blank, then keep whatever is already in the assemblyinfo file.
+            if (getVersion() != '') {
```

Comment:
could move the empty check to the replace method no ?

Reply:
I moved the checks into the replace function.  they're much better there anyway.  I'm not confident of the syntax for "not null and not empty" though.